src/algebra/curve.rs

```diff
@@ -158,44 +128,55 @@ impl StarkPoint {
     }
 
     /// Deserialize a point from a byte buffer
-    pub fn from_bytes(bytes: &[u8]) -> Result<StarkPoint, SerializationError> {
-        let point = StarkPointInner::deserialize_compressed(bytes)?;
-        Ok(StarkPoint(point))
+    pub fn from_bytes(bytes: &[u8]) -> Result<CurvePoint<C>, SerializationError> {
+        let point = C::deserialize_compressed(bytes)?;
+        Ok(CurvePoint(point))
+    }
+
+    /// Get the number of bytes needed to represent a point, this is exactly the number of bytes
+    /// for one base field element, as we can simply use the x-coordinate and set a high bit for the `y`
```

Comment:
Is this true for all different types of curves?

Reply:
Not universally, it is true for basically any curve that could be used here.